### PR TITLE
Added node metadata export for gltf2

### PR DESCRIPTION
**Pull request by bengtr**
glTF2Importer reads node metadata but glTF2Exporter did not write node metadata. Added node metadata export of all aiMetadataType types except AI_AIVECTOR3D.

### CODE REVIEW

**Comment by kimkulling** [comment]
You need to rebase your branch to the latest master. I am not allowed to update it.

File: code/AssetLib/glTF2/glTF2Asset.h [hunk 1]
```diff
@@ -376,6 +376,23 @@ struct CustomExtension {
     }
 };
 
+//! Represents metadata in an glTF2 object
+struct Extras {
+    std::vector<CustomExtension> mValues;
+
+    inline bool HasExtras() const {
+        return mValues.size() != 0;
```
Comment (kimkulling) [inline code/AssetLib/glTF2/glTF2Asset.h:384]: What about using empty?

File: code/AssetLib/glTF2/glTF2Asset.h [hunk 2]
```diff
@@ -376,6 +376,23 @@ struct CustomExtension {
...
+    }
+
+    Extras() = default;
+    ~Extras() = default;
+
+    Extras(const Extras &other) :
```
Comment (kimkulling) [inline code/AssetLib/glTF2/glTF2Asset.h:390]: I guess this copy constructor will be generated automatically by the compiler. So if you are not interested in having a special implementation feel free to remove it.

File: code/AssetLib/glTF2/glTF2Asset.inl [hunk 3]
```diff
@@ -139,6 +139,18 @@ inline CustomExtension ReadExtensions(const char *name, Value &obj) {
     return ret;
 }
 
+inline Extras ReadExtras(Value &obj) {
+    Extras ret;
+
+    ret.mValues.reserve(obj.MemberCount());
+    for (auto it = obj.MemberBegin(); it != obj.MemberEnd(); ++it) {
+        auto &val = it->value;
+        ret.mValues.push_back(ReadExtensions(it->name.GetString(), val));
```
Comment (kimkulling) [inline code/AssetLib/glTF2/glTF2Asset.inl:148]: Consider using emplace_back instead.

**Review by kimkulling — CHANGES_REQUESTED** [review]
PLease check my review comments. Rest looks fine for me.

**Comment by bengtr** [comment]
Added suggested changes and hopefully fixed g++/clang build complaints.

File: code/AssetLib/glTF2/glTF2Asset.inl (hunk 3 above)
Comment (bengtr) [reply code/AssetLib/glTF2/glTF2Asset.inl]: Hit the resolve conversation button by accident. Change is added and pushed to branch.

**Comment by bengtr** [comment]
Sorry, tried to resolve the requested changes status (not that familiar with the process) and I think I messed up the existing approval.

**Comment by kimkulling** [comment]
Not to worry, I will do this manually on this weekend!

**Comment by kimkulling** [comment]
Will be merged by https://github.com/assimp/assimp/pull/5109

**Pull request closed without merging**